tests/mesh/mesh_stitch.C

```diff
@@ -228,6 +224,169 @@ public:
+                                       HEX27);
+
+    // Make a copy
+    auto mesh1 = mesh0->clone();
+
+    // Shift copy one unit to the right
+    MeshTools::Modification::translate(*mesh1, /*x-dir*/1.0);
+
+    // For both meshes:
+    // .) Put odd-numbered Elems in elmset 1
+    // .) Put even-numbered Elems in elemset 2
+    // We use the trivial encoding: elemset id == elemset code for simplicity
+    auto place_elems = [](MeshBase & mesh)
+      {
+        unsigned int elemset_index =
+          mesh.add_elem_integer("elemset_code", /*allocate_data=*/true);
+
+        mesh.add_elemset_code(/*code=*/1, /*set_ids*/{1});
+        mesh.add_elemset_code(/*code=*/2, /*set_ids*/{2});
+
+        for (const auto & elem : mesh.element_ptr_range())
+          {
+            if (elem->id() % 2) // id odd
+              elem->set_extra_integer(elemset_index, 1);
+            else // id even
+              elem->set_extra_integer(elemset_index, 2);
+          }
+      };
+
+    place_elems(*mesh0);
+    place_elems(*mesh1);
+
+    // Before stitching, change the elemset codes on mesh1 so they
+    // don't overlap with the codes on mesh0.
+    mesh1->change_elemset_code(/*old*/1, /*new*/3); // 1 -> 3
+    mesh1->change_elemset_code(/*old*/2, /*new*/4); // 2 -> 4
+
+    // Before stitching, change the elemset ids on mesh1 so they
+    // don't overlap with the elemset ids on mesh0.
+    mesh1->change_elemset_id(/*old*/1, /*new*/100); // 1 -> 100
+    mesh1->change_elemset_id(/*old*/2, /*new*/200); // 2 -> 200
+
+    // Stitch the meshes together at the indicated boundary ids
+    mesh0->stitch_meshes(dynamic_cast<UnstructuredMesh &>(*mesh1),
+                         /*this boundary=*/2,
+                         /*other boundary=*/4,
+                         TOLERANCE,
+                         /*clear_stitched_boundary_ids=*/true,
+                         /*verbose=*/true,
+                         /*use_binary_search=*/false,
+                         /*enforce_all_nodes_match_on_boundaries=*/false);
+
+    // Number of elements in each mesh pre-stitch
+    dof_id_type n_elem_prestitch = Utility::pow<3>(ps);
+
+    // mesh0 should contain 2 * ps**3 total elements after stitching
+    CPPUNIT_ASSERT_EQUAL(static_cast<dof_id_type>(2 * n_elem_prestitch), mesh0->n_elem());
+
+    // Check that the stitched mesh still stores "elemset_code" in the
+    // same index (0) as it was before the meshes were stitched.
+    unsigned int elemset_index = mesh0->get_elem_integer_index("elemset_code");
+    CPPUNIT_ASSERT_EQUAL(0u, elemset_index);
+
+    // Check that the stitched mesh has merged elemset codes and ids as expected
+    MeshBase::elemset_type id_set_to_fill;
+    for (const auto elemset_code : {1, 2, 3, 4})
+      {
+        mesh0->get_elemsets(elemset_code, id_set_to_fill);
+
+        // Should be one elemset id in each set
+        CPPUNIT_ASSERT_EQUAL(static_cast<std::size_t>(1), id_set_to_fill.size());
+
+        switch (elemset_code)
```

Comment:
There's about 30 lines here that could be replaced with something like:
```
const MeshBase::elemset_type code_to_type[] = {0,1,2,100,200};
CPPUNIT_ASSERT(id_set_to_fill.count(code_to_type[elemset_code]));
```

Right?

But that's the worst nitpick I have, and it's in a unit test so I hardly care if it's cleaned up.

Reply:
Oh, yeah that would work because of the particular elemset_codes I chose, huh. I can make that change once it looks like there aren't any more warnings to address.